frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form/ocs-storage-class-form.tsx

```diff
@@ -543,17 +373,134 @@ export const StorageClassEncryptionKMSID: React.FC<ProvisionerProps> = ({
+    }
+  }, [csiConfigMap, csiConfigMapLoaded, csiConfigMapLoadError, setIsExistingKms, setEncryptionId]);
+
+  const updateKMS = async () => {
+    setInProgress(true);
+    const allServiceNames = csiKmsDetails ? Object.keys(csiKmsDetails?.data) : [];
+    if (
+      (allServiceNames.length && allServiceNames.indexOf(state.kms.name.value) === -1) ||
+      !csiKmsDetails
+    ) {
+      try {
+        const promises: Promise<K8sResourceKind>[] = createCsiKmsResources(
+          state.kms,
+          !!csiKmsDetails,
+        );
+        await Promise.all(promises).then(() => {
+          setIsExistingKms(true);
+          setEncryptionId(state.kms.name.value);
+        });
+        setErrorMessage('');
+      } catch (error) {
+        setErrorMessage(error.message);
+      }
+    } else {
+      setErrorMessage(
+        t('ceph-storage-plugin~KMS service {{value}} already exist', {
+          value: state.kms.name.value,
+        }),
+      );
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [csiConfigMap]);
+    setInProgress(false);
+  };
+
+  if (!csiConfigMapLoaded && !csiConfigMapLoadError) {
+    return <StatusBox loadError={csiConfigMapLoadError} loaded={csiConfigMapLoaded} />;
+  }
+  return (
+    <Form className="ocs-storage-class-encryption__form--padding">
+      <FormGroup fieldId="rbd-sc-kms-connection-selector">
+        <div id="rbd-sc-kms-connection-selector">
+          <Radio
+            label={t('ceph-storage-plugin~Choose existing KMS connection')}
+            name="kms-selection"
+            id="choose-existing-kms-connection"
+            className="ocs-storage-class-encryption__form-radio"
+            onClick={() => setIsExistingKms(true)}
+            checked={isExistingKms}
+          />
+          {isExistingKms && (
+            <ExistingKMSDropDown
+              csiConfigMap={csiKmsDetails}
+              serviceName={serviceName}
+              setEncryptionId={setEncryptionId}
+            />
+          )}
+          <Radio
+            label={t('ceph-storage-plugin~Create new KMS connection')}
+            name="kms-selection"
+            id="create-new-kms-connection"
+            className="ocs-storage-class-encryption__form-radio"
+            onClick={() => setIsExistingKms(false)}
+            checked={!isExistingKms}
+          />
+          {!isExistingKms && (
+            <Card isFlat className="ocs-storage-class-encryption__card">
+              <KMSConfigure
+                state={state}
+                dispatch={dispatch}
+                className="ocs-storage-class-encryption"
+              />
+              <div className="ocs-install-kms__save-button">
+                <ButtonBar errorMessage={errorMessage} inProgress={progress}>
+                  <ActionGroup>
+                    <Button
+                      variant="secondary"
+                      onClick={updateKMS}
+                      isDisabled={!scKmsConfigValidation(state.kms)}
```

Comment:
```suggestion
                      isDisabled={!state.isHandled}
```

Reply:
this is not give the desired results... state.isHandled is set if 
```
kms.token?.valid && kms.token.value !== '' && scKmsConfigValidation(kms);
```
but for storage-class KMS we do not have `kms.token.value` thus "save" button will always be disabled... here we only need `scKmsConfigValidation(state.kms)`

Reply:
Ah! yes. my bad :)